util/chunk/disk.go

```diff
@@ -83,11 +85,24 @@ func NewListInDisk(fieldTypes []*types.FieldType) *ListInDisk {
 	return l
 }
 
+// Len returns the number of rows in ListInDisk
+func (l *ListInDisk) Len() int {
+	return l.numRowsInDisk
```

Comment:
Could we count the `l.offsets` to get the total number rows?

Reply:
No, l.offsets are a two dimensional array, you can't count the number of rows unless you iterate through the first dimension, however doing so introduces complexity concern, turning a O(1) operation into O(number of chunks), therefore I prefer to keep it.